### PR TITLE
HIVE-26354: Support expiring snapshots on iceberg table

### PR DESCRIPTION
**Pull request by pvary**
### What changes were proposed in this pull request?
Adds the following syntax for expiring the Iceberg table snapshots:

ALTER TABLE test_table EXECUTE expire_snapshots('timestamp');

### Why are the changes needed?
Additional feature for Iceberg tables

### Does this PR introduce _any_ user-facing change?
New syntax

### How was this patch tested?
Added unit test

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/execute/AlterTableExecuteAnalyzer.java [hunk 1]
```diff
@@ -72,14 +75,26 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
...
+        spec = new AlterTableExecuteSpec(ROLLBACK, new RollbackSpec(TIME, time.toEpochMilli()));
       } else {
-        spec = new AlterTableExecuteSpec(ROLLBACK, new AlterTableExecuteSpec.RollbackSpec(VERSION,
+        spec = new AlterTableExecuteSpec(ROLLBACK, new RollbackSpec(VERSION,
             Long.valueOf(child.getText())));
       }
       desc = new AlterTableExecuteDesc(tableName, partitionSpec, spec);
     }
 
+    if (HiveParser.KW_EXPIRE_SNAPSHOTS == executeCommandType.getType()) {
```
Comment (lcspinter) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/table/execute/AlterTableExecuteAnalyzer.java:86]: nit: from a readability perspective else if statement would be better

File: ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableExecuteSpec.java [hunk 2]
```diff
@@ -25,13 +25,15 @@
  * The following operation are supported
```
Comment (lcspinter) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableExecuteSpec.java:25]: nit: operations

**Review by lcspinter — APPROVED** [review]
LGTM, just some minor comments

File: ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableExecuteSpec.java (hunk 2 above)
Comment (pvary) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableExecuteSpec.java]: done

**Pull request merged**